tests/SmartRange.test.ts

```diff
@@ -0,0 +1,195 @@
+import { describe, expect, test } from "vitest";
+import { SmartRange } from "../src";
+
+describe("SmartRange", () => {
```

Comment:
Maybe a test with a really big SmartRange could be interesing for performance tests